ginivision/src/main/java/net/gini/android/vision/analysis/AnalysisFragmentImpl.java

```diff
@@ -110,15 +133,111 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         stopScanAnimation();
     }
 
     public void onStart() {
+        showHints();
+    }
+
+    private void showHints() {
+
+        final int containerHeight = getContainerHeight();
+        mHints = generateRandomHintsList();
+
+        mRunnable = new Runnable() {
+            @Override
+            public void run() {
+                mHintAnimation = getSlideDownAnimation(containerHeight);
+                mHintAnimation.start();
+            }
+        };
+        mHandler.postDelayed(mRunnable, HINT_START_DELAY);
+    }
+
+    @NonNull
+    private ViewPropertyAnimatorCompat getSlideDownAnimation(final int containerHeight) {
+        return ViewCompat.animate(mHintContainer)
+                .translationY(containerHeight)
+                .setDuration(HINT_ANIMATION_DURATION)
+                .setListener(new ViewPropertyAnimatorListener() {
+                    @Override
+                    public void onAnimationStart(final View view) {
+                    }
+
+                    @Override
+                    public void onAnimationEnd(final View view) {
+                        setNextHint();
+                        mHintAnimation = getSlideUpAnimation();
+                        mHintAnimation.start();
+                    }
+
+                    @Override
+                    public void onAnimationCancel(final View view) {
+                    }
+                });
     }
 
-    public void onStop() {
+    @NonNull
+    private ViewPropertyAnimatorCompat getSlideUpAnimation() {
+        return ViewCompat.animate(mHintContainer)
+                .translationY(0)
+                .setDuration(HINT_ANIMATION_DURATION)
+                .setListener(new ViewPropertyAnimatorListener() {
+                    @Override
+                    public void onAnimationStart(final View view) {
+                    }
+
+                    @Override
+                    public void onAnimationEnd(final View view) {
+                        mHandler.postDelayed(mRunnable, HINT_CYCLE_INTERVAL);
+                    }
+
+                    @Override
+                    public void onAnimationCancel(final View view) {
+                    }
+                });
+    }
+
+    private int getContainerHeight() {
+        final Activity activity = mFragment.getActivity();
+        final Display display = activity.getWindowManager().getDefaultDisplay();
+        final Point size = new Point();
+        display.getSize(size);
+        return size.y;
+    }
+
+    private void setNextHint() {
+        final AnalysisHint nextHint= getNextHint();
+        final Context context = mFragment.getActivity();
+        mHintImageView.setImageDrawable(ContextCompat.getDrawable(context, nextHint.getDrawableResource()));
+        mHintTextView.setText(nextHint.getTextResource());
+    }
+
+    private AnalysisHint getNextHint() {
+        if(mHints.isEmpty()) {
+            mHints = generateRandomHintsList();
+        }
+        return mHints.remove(0);
```

Comment:
By reshuffling after all hints were shown we could end up showing the same hint twice one after the other. Wouldn't it be enough to shuffle it once and then loop through the list?